app/models/qdm/basetypes/interval.rb

```diff
@@ -32,6 +32,27 @@ def shift_dates(seconds)
       self
     end
 
+    def shift_years(year_shift)
+      if (@low.is_a? DateTime) || (@low.is_a? Time)
+        low_shift = @low.year + year_shift
+        @low = if @low.month == 2 && @low.day == 29 && !Date.leap?(low_shift)
+                 @low.change(year: low_shift, day: 28)
+               else
+                 @low.change(year: low_shift)
+               end
+      end
+      if (@high.is_a? DateTime) || (@high.is_a? Time)
+        high_shift = @high.year + year_shift
+        @high = if @high.month == 2 && @high.day == 29 && !Date.leap?(high_shift)
+                  @high.change(year: high_shift, day: 28)
+                else
+                  @high.change(year: high_shift)
+                end
+        @high = @high.year > 9999 ? @high.change(year: 9999) : @high
```

Comment:
We should add a similar line to the low block to make the lowest possible year to be 1 (and a unit test to show that it does so)

Reply:
Yep that makes sense. I also added this logic to `data_element.rb` as well.